File: sorting-algos/shell_sort/shell_sort.py
#shell sort starts with unsorted array
#selects one element and moves it multiple positions at a time. 
# similiar to insertion sort

#first compute gap n/2 (n = number of elements)

#Start from first element, compare it to  element at gap left and right

def shell_sort(array):
    gap = len(array) // 2
    n = len(array) - 1

    while gap > 0:
        i = gap
        while i < len(array):
            temp = array[i]
            j = i - gap
            while j >= 0 and array[j] > temp:
                array[j + gap] = array[j]
                j = j - gap
                array[j + gap] = temp
            i = i + 1
        gap = gap // 2

    return array

array = [ 3,5,2,6,4,5,7,8]
shell_sort(array)
print(array)

File: sorting-algos/shell_sort/shell_sort.js
const shell_sort = array => {
    let n = array.length

    for(let gap = Math.floor(n/2); gap > 0; gap = Math.floor(n/2)){
        //test
        for(let i = gap; i < n; i++){
            let gvalue= array[i]
            let j;
			for (j = i; j >= gap && array[j-gap] > gvalue; j-=gap)  {
				array[j] = array[j-gap];
			}
            array[j] = gvalue
        }
    }
}

array = [4,5,3,6,2,7,5,8,6]

console.log(array)

shell_sort(array)

console.log(array)